### ISSUE TITLE
Setup all necessary onchain context in `docker-compose`

### ISSUE THREAD

**Issue by sistemd** [cowprotocol/services#850]
Our `docker-compose` currently sets up a local Ethereum environment via `hardhat`, but doesn't deploy any of the necessary contracts or other setup. This is confusing to users. Recently, a person was trying to work on `services` locally and ended up asking us how to deploy the contracts. It'd be best if `docker-compose` did all this work automatically so that the users could simply run the services and use them locally end-to-end.

This along with other `docker-compose` functionality should be documented in our `README.md` file.

**Comment by MartinquaXD** [comment]
We also set up hardhat in a very particular way for the e2e tests (quoted from the readme):
```
3. Create hardhat.config.js in the directory you installed hardhat in with following content:

module.exports = {
    networks: {
        hardhat: {
            initialBaseFeePerGas: 0,
            accounts: {
                accountsBalance: "1000000000000000000000000"
            }
        }
    }
};
```
I already ran e2e tests with this `docker-compose` setup so I think it's not 100% required but in any case it would be good to keep `README.md`, `docker-compose.yaml` and the [pull request workflow](https://github.com/cowprotocol/services/blob/main/.github/workflows/pull-request.yaml#L78-L87) in sync.

Alternatively when this issue is completed and `docker-compose up` also sets up everything in the local test node we might as well drop the whole `Development Setup` section in the `README.md` in favor of `docker-compose up` leaving fewer things to keep up-to-date. I'm not sure if there are any use cases we care about that are not covered by the docker container at that point.
